### PR TITLE
feat(agent): openapi specification for agent

### PR DESCRIPTION
**Pull request by dscabral**
Openapi spec to Agent on fleet management

### CODE REVIEW

File: fleet/api/openapi.yaml [hunk 1]
```diff
@@ -103,6 +103,125 @@ paths:
...
+        '400':
+          description: Failed due to malformed query parameters.
+        '401':
+          description: Missing or invalid access token provided.
+        '404':
+          description: A non-existent entity request.
+        '500':
+          $ref: "#/components/responses/ServiceErrorRes"
+    post:
+      summary: 'Create a new Agent'
```
Comment (weyrick) [inline fleet/api/openapi.yaml:132]: this is missing parameters/Authorization i think?

File: fleet/api/openapi.yaml [hunk 2]
```diff
@@ -178,7 +311,15 @@ components:
       required: true
     AgentGroupId:
       name: id
-      description: Unique Agent Group identifier.
+      description: Unique Agent identifier.
```
Comment (weyrick) [inline fleet/api/openapi.yaml:314]: "Agent Group"

File: fleet/api/openapi.yaml [hunk 3]
```diff
@@ -267,6 +421,55 @@ components:
...
       required:
         - agent_groups
+    AgentUpdateReqSchema:
+      type: object
+      properties:
+        name:
+          type: string
+          description: A unique name label
+          example: eu-agents
+        agent_tags:
```
Comment (weyrick) [inline fleet/api/openapi.yaml:431]: `agent_tags` is the one that is read only from the API, and set from the agent itself. So this should be `orb_tags` instead

File: fleet/api/openapi.yaml [hunk 4]
```diff
@@ -267,6 +421,55 @@ components:
           description: Maximum number of items to return in one page.
       required:
         - agent_groups
+    AgentUpdateReqSchema:
+      type: object
+      properties:
+        name:
+          type: string
+          description: A unique name label
+          example: eu-agents
```
Comment (weyrick) [inline fleet/api/openapi.yaml:430]: Let's make this `my-agent1`

File: fleet/api/openapi.yaml [hunk 5]
```diff
@@ -267,6 +421,55 @@ components:
...
+      type: object
+      required:
+        - name
+        - tags
+      properties:
+        name:
+          type: string
+          description: A unique name label
+          example: eu-agents
+        agent_tags:
```
Comment (weyrick) [inline fleet/api/openapi.yaml:447]: again swap for `orb_tags`

File: fleet/api/openapi.yaml [hunk 6]
```diff
@@ -267,6 +421,55 @@ components:
...
+    AgentCreateReqSchema:
+      type: object
+      required:
+        - name
+        - tags
+      properties:
+        name:
+          type: string
+          description: A unique name label
+          example: eu-agents
```
Comment (weyrick) [inline fleet/api/openapi.yaml:446]: again `my-agent1`

File: fleet/api/openapi.yaml [hunk 7]
```diff
@@ -328,3 +531,64 @@ components:
...
+          description: A time stamp of the last heartbeat that was received
+    AgentValidateObjSchema:
+      type: object
+      required:
+        - id
+      properties:
+        name:
+          type: string
+          description: A unique name label
+          example: eu-agent
```
Comment (weyrick) [inline fleet/api/openapi.yaml:588]: same changes here

**Review by weyrick — CHANGES_REQUESTED** [review]

File: fleet/api/openapi.yaml (hunk 1 above)
Comment (dscabral) [reply fleet/api/openapi.yaml:132]: Done, i was trying to use the security schema, i forgot to rollback.

File: fleet/api/openapi.yaml (hunk 2 above)
Comment (dscabral) [reply fleet/api/openapi.yaml:314]: Done

File: fleet/api/openapi.yaml (hunk 4 above)
Comment (dscabral) [reply fleet/api/openapi.yaml:430]: Done

File: fleet/api/openapi.yaml (hunk 3 above)
Comment (dscabral) [reply fleet/api/openapi.yaml:431]: Done

File: fleet/api/openapi.yaml (hunk 6 above)
Comment (dscabral) [reply fleet/api/openapi.yaml:446]: Done

File: fleet/api/openapi.yaml (hunk 7 above)
Comment (dscabral) [reply fleet/api/openapi.yaml:588]: Done

File: fleet/api/openapi.yaml (hunk 5 above)
Comment (dscabral) [reply fleet/api/openapi.yaml]: Done

**Review by weyrick — APPROVED** [review]

**Pull request merged**